russstation/code/game/objects/items/miscellaneous.dm

```diff
@@ -0,0 +1,172 @@
+		qdel(src)
+		new /obj/item/clothing/suit/caution(L, 1)
+		new /obj/item/janiupgrade(L, 1)
+
+	if(istype(I, /obj/item/assembly/prox_sensor))
+		to_chat(user, "<span class='notice'>You add a [I.name] to the floor buffer on the [src.name].<span>")
+		qdel(I)
+		qdel(src)
+		var/obj/item/S = /obj/item/clothing/suit/caution/slippery
+		if(prob(1)) //1% chance for a name
+			S.name = pick("Ms. Lippy", "Mr. Walky", "Monitor Hallsky")
+
+		new S(L, 1)
+
+//old signs (only found in maint spawners)
+/obj/item/caution/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I, /obj/item/janiupgrade))
+		to_chat(user, "<span class='warning'>The [src.name] is too antiquated to fit a [I.name], try a newer sign.<span>")
+
+
+//end of construction code
+
+//Wetmore Slippery Sign
+//Constructable by janitors with upgrades from RND and sensors from cargo/robotics
+//Slips people who don't heed the warning
+
+//Wetmore Slippery Sign (traitor item)
+//purchasable from traitor uplink by janitors (7tc for 4)
+//in addition to slipping, also animates and kicks people around
+
+/obj/item/clothing/suit/caution/slippery
+	slot_flags = null //cannot be worn- or else it breaks stuff
+
+	var/willSlip = 0 //0 - disabled || 1 - enabled 
+	var/lastSlip = 0 //last time the sign slipped someone (time)
+	var/slipCooldown = 50 //cooldown (~5 seconds default)
+	var/clowningAround = 0 //0 - untainted || 1 - clown got it
+	var/evilSign = 0 //is the sign a traitor?
+	var/mob/living/carbon/boss = null //used so animated signs don't attack the janitor
+
+/obj/item/clothing/suit/caution/slippery/syndicate
+	evilSign = 1 //signs purchased from the uplink
+
+/obj/item/clothing/suit/caution/slippery/examine(mob/user)
+	. = ..()
+	if(isobserver(user) || user.mind.assigned_role == "Janitor") //janitors and ghosts can see that it's a slippery sign
+		. += "<span class='notice'>This [src.name] is outfitted with an experimental sprayer.</span>"
+		if(clowningAround || evilSign)
+			. += "<span class='warning'>It's been tampered with.</span>"
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY)) //clowns know lube when they see it
+		. += "<span class='notice'>This [src.name] has great potential for pranks.</span>"
+
+
+//when used by a janitor: toggles between active and disabled, when used by a clown, pranks ensue
+/obj/item/clothing/suit/caution/slippery/attack_self(mob/user)
+	if(!proximity_monitor)
+		proximity_monitor = new(src, 1) //initializes the proximity: (source, range)
+
+	if(user.mind.assigned_role == "Janitor") //only janitors can interact with it normally
+		boss = user
+		willSlip = !willSlip
+		if(clowningAround) //so you can reset the sign if a clown messes with it
+			clowningAround = 0
+			slipCooldown = 50 
+
+		if(willSlip)
+			to_chat(user, "<span class='notice'>The [src.name] will now slip anyone running past.<span>")
+		else
+			to_chat(user, "<span class='notice'>The [src.name] will no longer slip passerbys.<span>")
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY)) //clowns.
```

Comment:
See the above comment, but include that being clumsy and being a janitor aren't technically mutually exclusive. While this would still work as intended, if giving someone clumsy made this work for them, but *only if they're not a janitor* would be unexpected.

Reply:
moved the clumsy check to before the janitor check to account for this